### PR TITLE
Fix Simple Pylance typing error

### PR DESCRIPTION
**Pull request by Quasimewdo**

### CODE REVIEW

**Review by beeender — COMMENTED** [review]
Can the same warning be reported by pyright? It's better we have the linter set for CI so a similar issue can be caught in the future.

To change the configuration for pyright, change https://github.com/greenplum-db/GreenplumPython/blob/master/pyproject.toml

And here are some configurations for pyright: https://github.com/microsoft/pyright/blob/main/docs/configuration.md

Otherwise, maybe we can set up pylance on CI as well. Let me know if you need help on this.

**Comment by Quasimewdo** [comment]
> Can the same warning be reported by pyright? It's better we have the linter set for CI so a similar issue can be caught in the future.
> 
> To change the configuration for pyright, change https://github.com/greenplum-db/GreenplumPython/blob/master/pyproject.toml
> 
> And here are some configurations for pyright: https://github.com/microsoft/pyright/blob/main/docs/configuration.md
> 
> Otherwise, maybe we can set up pylance on CI as well. Let me know if you need help on this.

We can add this `typeCheckingMode = "strict"` in `pyproject.toml`

**Comment by beeender** [comment]
> We can add this typeCheckingMode = "strict" in pyproject.toml

I tried that at the beginning, but that was "too" strict at that time. Will it pass now with strict mode?

**Comment by beeender** [comment]
> Well not yet, there are still lots of reports
That might be too much ...
If possible, we shall only enable the configs we care

**Comment by beeender** [comment]
```
# pyproject.toml
[tool.pyright]
include = [
    "greenplumpython",
    "setup.py"
]
strict =[
    "greenplumpython",
]
```

Will cause more lint failures than what we fixed in this PR. Maybe we can enable some of the strict rules but not all of them. 

@Quasimewdo Let me know if you need help on this.

**Comment by Quasimewdo** [comment]
I have added `typeCheckingMode = "strict"` in `pyproject.toml`, and set some report o `false`. 
I have also ignored some current unsolvable type error report by adding `#type : ignore` in code.
@beeender @xuebinsu

**Review by beeender — COMMENTED** [review]
LGTM!
Need python experts on this @xuebinsu @yihong0618 

I am THE PY IDIOT

File: greenplumpython/util.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    def get(self, fn, annotations, *args):
+        """get returns the matching function from the virtual namespace.
+        return None if it did not fund any matching function.
+        """
+        func = Function(fn)
+        return self.function_map.get(func.key(args=args, annotations=annotations))
+
+
+def overload(fn):
```
Comment (Quasimewdo) [inline greenplumpython/util.py:91]: This can't overpass `Pyright` error

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -25,6 +25,10 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Set up Node # for pyright
+        uses: actions/setup-node@v1
+        with:
+          node-version: '12'
```
Comment (yihong0618) [inline .github/workflows/lint.yml:31]: need to change with the pyright version

**Review by yihong0618 — APPROVED** [review]

**Pull request merged**